script.js

```diff
@@ -1,12 +1,602 @@
+    <div id="allShowsContainer" class="container-fluid show-content">
+      <div id="freeSearchResult" class="row" style="display: none">
+        <h2 class="row shows-genre-headers">Search Result</h2>
+        <div id="freeSearchResultShows" class="card-group row flex-nowrap row-cols-1 row-cols-md-2 row-cols-lg-3 g-4">
+        </div>
+      </div>
+      <h2 class="row shows-genre-headers">Top Rated Shows</h2>
+      <div id="topRatedShows" class="shows-genre-card-group card-group row flex-nowrap row-cols-1 row-cols-md-2 row-cols-lg-3 g-4">
+      </div>
+      <h2 class="row shows-genre-headers">Drama Shows</h2>
+      <div id="dramaShows" class="shows-genre-card-group card-group row flex-nowrap row-cols-1 row-cols-md-2 row-cols-lg-3 g-4">
+      </div>
+      <h2 class="row shows-genre-headers">Comedy Shows</h2>
+      <div id="comedyShows" class="shows-genre-card-group card-group row flex-nowrap row-cols-1 row-cols-md-2 row-cols-lg-3 g-4">
+      </div>
+      <h2 class="row shows-genre-headers">Crime Shows</h2>
+      <div id="crimeShows" class="shows-genre-card-group card-group row flex-nowrap row-cols-1 row-cols-md-2 row-cols-lg-3 g-4">
+      </div>
+      <h2 class="row shows-genre-headers">Action Shows</h2>
+      <div id="actionShows" class="shows-genre-card-group card-group row flex-nowrap row-cols-1 row-cols-md-2 row-cols-lg-3 g-4">
+      </div>
+      <h2 class="row shows-genre-headers">Sci-Fi Shows</h2>
+      <div id="scifiShows" class="shows-genre-card-group card-group row flex-nowrap row-cols-1 row-cols-md-2 row-cols-lg-3 g-4">
+      </div>
+    </div>
+    <div id="showContainer" class="container">
+      <div id="castListSection">
+        <h2 class="row cast-list-header">Cast</h2>
+        <div id="castList" class="card-group row flex-nowrap row-cols-1 row-cols-md-2 row-cols-lg-3 g-4" style='display:none'>
+        </div>
+      </div>
+      <h2 class="row episodes-header">Episodes</h2>
+      
+      <div id="seasonInputContainer" class="row" style="display:none">
+        <div class="col-md-4 offset-md-1">
+          <input id='input' type="text" class="input form-control" placeholder="search episodes">
+        </div>
+        
+        <div class="col-md-3">
+          <select class="form-select" id="seasonSelect">
+          </select>
+        </div>
+        <div class="col-md-3">
+          <select id='selectEpisode' name="" class="form-select">
+            <option value="">All Episodes</option>
+          </select>
+        </div>
+        <p id="epAmount" class="col-10 offset-md-1"></p>
+      </div>
+      <div id="seasonEpisodes" class="row show-content">
+      </div>
+    </div>
+  `;
+  rootElem.innerHTML = pageSetupHtml;
+
+})();
+
+function makeCastList(castList){
+  let castListDiv = document.getElementById('castList');
+  castListDiv.style.display = 'flex';
+  let castListDisplay = castList.map((cast) => {
+        let characterName = cast && cast.character ? cast.character.name : null;
```

Comment:
Another way you can check for a value before accessing could be to use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

Reply:
Thanks for this added optional chaining, even found out about the nullish coalescing operator condensed the code nicely